### PR TITLE
Api user

### PR DESCRIPTION
**Pull request by krisolchova**

### CODE REVIEW

File: web-app/src/App.js [hunk 1]
```diff
@@ -1,19 +1,34 @@
...
+import SignIn from './SignIn';
 
 function App() {
   return (
     <Provider store={store}>
-      <Header />
-      <SignUp />
+      <Router>
+        <Header />
+
```
Comment (MartinDawson) [inline web-app/src/App.js:17]: Remove these empty lines please

File: web-app/src/BondData.js [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+const useStyles1 = makeStyles((theme) => ({
+  root: {
+    flexShrink: 0,
+    marginLeft: theme.spacing(2.5),
+  },
+}));
+
+
+function BondData(props) {
+
```
Comment (MartinDawson) [inline web-app/src/BondData.js:22]: remove line

File: web-app/src/BondData.js [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+import { makeStyles } from '@material-ui/core/styles';
+import Table from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableCell from '@material-ui/core/TableCell';
+import TableContainer from '@material-ui/core/TableContainer';
+import TableHead from '@material-ui/core/TableHead';
+import TableRow from '@material-ui/core/TableRow';
+import Paper from '@material-ui/core/Paper';
+
+const useStyles1 = makeStyles((theme) => ({
```
Comment (MartinDawson) [inline web-app/src/BondData.js:13]: should be `useStyles` not `useStyles1`

File: web-app/src/BondData.js [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+      .then(response => {
+        console.log(response)
+        setBondData(response.data)
+      })
+      .catch(error => {
+        console.log(error)
+      })
+  }, [props.match.bondId])
+
+
```
Comment (MartinDawson) [inline web-app/src/BondData.js:38]: remove 2 lines here

File: web-app/src/BondData.js [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+function BondData(props) {
+
+  const classes = useStyles1();
+  const [bondData, setBondData] = useState();
+
+  useEffect(() => {
+    axios
+      .get(`/api/v1/bond/${props.match.bondId}`)
+      .then(response => {
+        console.log(response)
```
Comment (MartinDawson) [inline web-app/src/BondData.js:30]: remove console log

File: web-app/src/BondData.js [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+
+
+
+  return bondData ? (
+    <div>
+      <h1 className={classes.h1}> {bondData.Name} SUMMARY BOND DATA</h1>
+      <TableContainer component={Paper}>
+        <Table className={classes.table} aria-label="simple table">
+          <TableBody>
+
```
Comment (MartinDawson) [inline web-app/src/BondData.js:46]: remove empty line

File: web-app/src/BondData.js [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+      .catch(error => {
+        console.log(error)
+      })
+  }, [props.match.bondId])
+
+
+
+  return bondData ? (
+    <div>
+      <h1 className={classes.h1}> {bondData.Name} SUMMARY BOND DATA</h1>
```
Comment (MartinDawson) [inline web-app/src/BondData.js:42]: `> {bondData.Name}` -> `>{bondData.Name}`

File: web-app/src/BondData.js [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+      <TableContainer component={Paper}>
+        <Table className={classes.table} aria-label="simple table">
+          <TableBody>
+
+            <TableRow>
+              <TableCell component="th" scope="row">
```
Comment (MartinDawson) [inline web-app/src/BondData.js:48]: use an array map of objects instead to get rid of all this duplication.

File: web-app/src/Header.js [hunk 9]
```diff
@@ -77,9 +82,24 @@ const useStyles = makeStyles((theme) => ({
       display: 'none',
     },
   },
+  homeIcon: {
+    marginRight: 4,
+    fontSize: 20,
+
```
Comment (MartinDawson) [inline web-app/src/Header.js:88]: remove new line

File: web-app/src/Header.js [hunk 10]
```diff
@@ -77,9 +82,24 @@ const useStyles = makeStyles((theme) => ({
...
+  },
+  homeButton: {
+    lineHeight: 1.6,
+    marginRight: '10px',
+  },
+  sigInButton: {
```
Comment (MartinDawson) [inline web-app/src/Header.js:94]: unused property

File: web-app/src/Header.js [hunk 11]
```diff
@@ -77,9 +82,24 @@ const useStyles = makeStyles((theme) => ({
...
     },
   },
+  homeIcon: {
+    marginRight: 4,
+    fontSize: 20,
+
+  },
+  homeButton: {
+    lineHeight: 1.6,
+    marginRight: '10px',
```
Comment (MartinDawson) [inline web-app/src/Header.js:92]: can just be 10 without the string or px like on line 86

File: web-app/src/Header.js [hunk 12]
```diff
@@ -77,9 +82,24 @@ const useStyles = makeStyles((theme) => ({
...
+  },
+  sigInButton: {
+
+  },
+  registerButton: {
+    marginRight: '10px',
```
Comment (MartinDawson) [inline web-app/src/Header.js:98]: same as above

File: web-app/src/Header.js [hunk 13]
```diff
@@ -104,6 +124,18 @@ export default function Header() {
     setMobileMoreAnchorEl(event.currentTarget);
   };
 
+  const [reDirectUser, setReDirectUser] = useState('')
+
+  const handleKeyDown = (e) => {
+    if (e.key === 'Enter') {
+      console.log('do validate');
```
Comment (MartinDawson) [inline web-app/src/Header.js:131]: remove console logs, not needed anymore cause of the push

File: web-app/src/Header.js [hunk 14]
```diff
@@ -162,71 +194,121 @@ export default function Header() {
...
+                color="inherit"
+                component={Link}
+                to="/"
+              > <b>BOND TRACKER</b>
+              </Button>
+            </Typography>
+            <div className={classes.search}>
+              <div className={classes.searchIcon}>
+                <SearchIcon />
+
```
Comment (MartinDawson) [inline web-app/src/Header.js:221]: remove empty new line

File: web-app/src/Header.js [hunk 15]
```diff
@@ -104,6 +124,18 @@ export default function Header() {
     setMobileMoreAnchorEl(event.currentTarget);
   };
 
+  const [reDirectUser, setReDirectUser] = useState('')
```
Comment (MartinDawson) [inline web-app/src/Header.js:127]: it should be called bondId or bondIsin or something more descriptive and setBondId or similar

File: web-app/src/Header.js [hunk 16]
```diff
@@ -162,71 +194,121 @@ export default function Header() {
...
+
+                  <Button
+                    className={classes.sigInButton}
+                    component={Link}
+                    to="/sign-in"
+                    variant="outlined"
+                    color="inherit">
+                    <b>SIGN IN</b>
+                  </Button>
+
```
Comment (MartinDawson) [inline web-app/src/Header.js:288]: new line remove

File: web-app/src/Header.js [hunk 17]
```diff
@@ -162,71 +194,121 @@ export default function Header() {
...
+              >
+                <MoreIcon />
+              </IconButton>
+            </div>
+          </Toolbar>
+        </AppBar>
+        {renderMobileMenu}
+        {renderMenu}
+      </div>
+
```
Comment (MartinDawson) [inline web-app/src/Header.js:308]: new line remove

File: web-app/src/HomePage.js [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+    fontSize: 30,
+  },
+
+  h1: {
+    textAlign: 'center',
+    color: '#3f51b5',
+    marginTop: 200,
+    fontSize: 50,
+  },
+
```
Comment (MartinDawson) [inline web-app/src/HomePage.js:42]: new lines remove in this file

File: web-app/src/HomePage.js [hunk 19]
```diff
@@ -0,0 +1,82 @@
...
+  },
+  fontSizeLarge: {
+    fontSize: '1.5em',
+  }
+}));
+
+function HomePage(props) {
+  const classes = useStyles();
+
+  const [inputUser, setInputUser] = useState('')
```
Comment (MartinDawson) [inline web-app/src/HomePage.js:54]: call it `bondId` or `bondIsin` or similar.

File: web-app/src/SignIn.js [hunk 20]
```diff
@@ -0,0 +1,123 @@
+import React, { useState } from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import Button from '@material-ui/core/Button';
+import TextField from '@material-ui/core/TextField';
+import { connect } from 'react-redux';
+import Axios from 'axios';
+
+
```
Comment (MartinDawson) [inline web-app/src/SignIn.js:8]: new lines

File: web-app/src/Header.js (hunk 15 above)
Comment (krisolchova) [reply web-app/src/Header.js:127]: fixed

File: web-app/src/HomePage.js (hunk 19 above)
Comment (krisolchova) [reply web-app/src/HomePage.js:54]: done

**Pull request merged**